tasks/tests/integration/test_upload_e2e.py

```diff
@@ -0,0 +1,194 @@
+import json
+from functools import partial
+from typing import Iterable
+from uuid import uuid4
+
+import pytest
+from shared.reports.types import ReportLine
+from shared.yaml import UserYaml
+from sqlalchemy.orm import Session
+
+from database.tests.factories import CommitFactory, RepositoryFactory
+from services.archive import ArchiveService
+from services.report import ReportService
+from tasks.tests.unit.test_upload_task import FakeRedis
+from tasks.tests.utils import hook_session, run_tasks
+from tasks.upload import upload_task
+
+
+# TODO: instead of `mock_redis`, can we use the real thing?
```

Comment:
I think we already use containers to run the tests in CI so it might not be so hard to spin up a redis instance there...
Apparently there's also (pytest-redis)[https://pypi.org/project/pytest-redis/] that might be useful

Reply:
I just removed all the mocking here, and I can just use the redis instance thats already running in docker which works fine.